internal/controller/utils/nim.go

```diff
@@ -189,12 +190,26 @@ func getNimRuntimes(runtimes []NimRuntime, page, pageSize int) ([]NimRuntime, er
 
 	runtimes = append(runtimes, mapNimCatalogResponseToRuntimeList(catRes)...)
 	if catRes.Params.Page < catRes.ResultPageTotal-1 {
-		return getNimRuntimes(runtimes, page+1, pageSize)
+		return getNimRuntimes(logger, runtimes, page+1, pageSize)
 	}
 
 	return runtimes, nil
 }
 
+func logApiResponse(logger logr.Logger, resp *http.Response) {
```

Comment:
for this func, I am not sure if it is really necessary to use the in context logger, instead, create a new logger here, it would avoid passing the logger as parameter everywhere.

Reply:
Why? We need to log to be consistent.

Reply:
The logger you are passing as parameter has nothing special: https://github.com/opendatahub-io/odh-model-controller/pull/383/files#diff-ac9aee3dd976038fa2ac5fc8ba9dddd1f98bb8bf71e7a924926351981a35f07aR46

so, why not instantiate it here inside this func and avoid passing the logger as parameter?

Reply:
@spolti You're looking at the hack script, see the previous comment, https://github.com/opendatahub-io/odh-model-controller/pull/383#discussion_r1975482886.

Check the code for the actual controller, the logger is proeperly instantiated by the controller, see: https://github.com/opendatahub-io/odh-model-controller/pull/383/files#diff-7617d7be763ad3db3310e0dcecb5978ebbc608f8ff2767970a01efe2fa941c70R206.

Reply:
Ah, got it, thanks, it seems I overlooked that :D